docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,177 @@
+
+**Plaintext files**: We'll use [html-to-text](https://github.com/html-to-text/node-html-to-text) to automatically generate plaintext versions of templates rendered to HTML. There may be a scenario in which the automatically-generated plaintext version does not look exactly how we'd like, in which case we can look into exporting both an MJML _and_ plaintext version of the email from the template file.
+
+**Localization**: We will upgrade the localization tool from GetText to [Fluent](https://github.com/projectfluent/fluent.js) since it's preferred by the l10n team and other FxA front-ends are using it. With our chosen templating option we can make use of Fluent's [`@fluent/dom`](https://github.com/projectfluent/fluent.js/tree/master/fluent-dom) package.
+
+[**Integration & QA**: During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.] **Vijay to edit**
+
+### Positive Consequences
+
+- Using MJML abstracts HTML email peculiarities away and handles responsiveness for us
+- Allows us to move away from inline CSS during development, and improves style reusability and consistency in template files
+- Using EJS allows us to write templates using JavaScript, removing the complexities of custom syntaxes like JSX and Mustache
+
+### Negative Consequences
+
+- The Storybook setup will be much more complex than a React and Tailwind solution
+- Introduces new dependencies, and MJML introduces a small learning curve
+- Our email templates have been battle-tested over the years, and this change could introduce potential new bugs across various clients
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache and ad-hoc inline styles
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
```

Comment:
@Yogita98 I feel like we address Les' comment under the "Cons". Maybe in Pros, we can go ahead and highlight that Mustache is designed to be simple and easy to read.